### DISCUSSION TITLE
Is it possible to show the keys pressed in my final GIF and use in Linux?

### DISCUSSION THREAD

**Discussion by SoyDiego** [NickeManarin/ScreenToGif#1254] [Q&A] [answered]
Hi @NickeManarin, thanks for this amazing app.
I have two questions: 
- Is it possible to show the keys pressed in my final GIF?
- Is it possible use in Linux?

Thank you!!

**Comment by pawlos** [comment, marked as answer]
@SoyDiego for the key pressed you can use  keystrokes option
![image](https://github.com/NickeManarin/ScreenToGif/assets/1296768/079ad011-1ef0-45dc-9bbd-ea27da64fd37)

**Reply by SoyDiego** [reply]
Thanks for your reply. I will check tomorrow.
And about Linux? Please tell me if possible use there and I will celebrate haha. I love this app

**Reply by NickeManarin** [reply]
I saw some person running the app on Linux, by using Wine.

Proper support for Linux distros will come with v3.

**Reply by SoyDiego** [reply]
Thanks both for your replies. I can't wait for that version.
Now I will try with wine If I can run it.
Do you have a estimate release date?

Thanks again!!

**Comment by MichaelWoodc** [comment]
> Thanks both for your replies. I can't wait for that version. Now I will try with wine If I can run it. Do you have a estimate release date?
> 
> Thanks again!!

As a perhaps poor workaround, you could also always be running your linux in a VM, or accessing it via VNC.  screentogif would need to be running as an administrator, and, a few functions don't work depending on your VM host, as some security options like credential guard are extremely difficult to bypass!  This is due to security reasons, like, if your host is compromised you can still securely use a VM without it stealing your passwords, for example.  Some good tools to use for linux include:
screenkey, this only shows the keys IIRC, to do the screenshots on linux you can write your own script, something similar to this (no idea if it will work, haven't tried it yet as I got screentogif to work from my vm host):

>

```
#!/bin/bash
while true; do
    # Wait for a mouse click or Enter key press
    xdotool selectwindow click 1 key Return

    # Capture the screenshot
    scrot '%Y-%m-%d-%H:%M:%S.png' -e 'mv $f ~/Screenshots/'

    # Sleep for a short duration to prevent rapid screenshots
    sleep 1
done
```
